### PR TITLE
re-factor(update-filter-behavior): Fix the `modifyFilters` function

### PR DESCRIPTION
**Pull request by ajtejankar**
Use lambdas and remove unnecessary conditionals.

### CODE REVIEW

File: t-update-filter-behavior.html [hunk 1]
```diff
@@ -188,7 +179,7 @@
     },
 
     _executeFunctionByName: function (functionName, context /*, args */) {
-        var args = [].slice.call(arguments).splice(2);
```
Comment (ajtejankar) [inline t-update-filter-behavior.html:191]: Do you really need `splice` here?. Slicing the arguments can be handled in `slice` call itself.

File: t-update-filter-behavior.html [hunk 2]
```diff
@@ -26,61 +26,52 @@
...
-                            for(var filterLebel in buildFilters[filterCategory]){
-                                if (filters[i].items[j].label.toLowerCase() == filterLebel.toLowerCase()) {
-                                    if (filters[i].items[j].additionalInfo && filters[i].items[j].additionalInfo.length) {
-                                        filters[i].items[j].additionalInfo[0].value = buildFilters[filterCategory][filterLebel];
-                                    }
-                                }
-                            }
-                        }
+        filters.forEach(function(filter) {
+            var buildFilter = buildFilters[filter.category];
```
Comment (ajtejankar) [inline t-update-filter-behavior.html:55]: We can get rid of the nested for loop iterating over object properties and directly see if there is a value for the key instead.

File: t-update-filter-behavior.html [hunk 3]
```diff
@@ -26,61 +26,52 @@
...
+
+            if (buildFilter && filter.type === "Range") {
+                filter.items[0].value = buildFilter.min.value
+                filter.items[0].displayValue = buildFilter.min.displayValue
+                filter.items[1].value = buildFilter.max.value
+                filter.items[1].displayValue = buildFilter.max.displayValue
+            }
+            else if (buildFilter && filter.type === "Options") {
+                filter.items.forEach(function(item) {
```
Comment (ajtejankar) [inline t-update-filter-behavior.html:64]: We can get rid of nested for loop and two similar for loops by directly accessing the property and merging the bodies of both for loops.

File: t-update-filter-behavior.html (hunk 1 above)
Comment (ganeshr43) [inline t-update-filter-behavior.html:191]: Yes, there is not need of splice here.

File: t-update-filter-behavior.html (hunk 3 above)
Comment (ganeshr43) [inline t-update-filter-behavior.html:64]: That's more precise :+1:

**Pull request merged**